### PR TITLE
Remove deprecated support for size/shape returning intIdxType

### PR DESCRIPTION
**Pull request by bradcray**
This continues the work started in #17628 to change .size and .shape
queries on ranges and domains to return values using `int` rather than
`intIdxType`.  It adds a new deprecation warning for using the
`sizeReturnsInt` config param that was introduced to choose between
return types, and a deprecation test to make sure we remove it in the
future.

### CODE REVIEW

File: modules/internal/ChapelDomain.chpl [hunk 1]
```diff
@@ -1262,26 +1262,13 @@ module ChapelDomain {
       for i in _value.dimIter(d, ind) do yield i;
     }
 
-   /* Return a tuple of :proc:`intIdxType` describing the size of each
-      dimension.  Note that this routine is in the process of changing
-      to return a tuple of ``int``, similar to the ``.size`` query on
-      ranges, domains, and arrays.  See :proc:`ChapelRange.range.size` for details.
+   /* Return a tuple of `int` describing the size of each
```
Comment (lydia-duncan) [inline modules/internal/ChapelDomain.chpl:1265]: I think this should be wrapped by two ` on either side, like the removed reference in the third line.

**Review by lydia-duncan — APPROVED** [review]
One minor documentation fix, otherwise this looks great to me!

File: modules/internal/ChapelDomain.chpl (hunk 1 above)
Comment (bradcray) [reply modules/internal/ChapelDomain.chpl:1265]: Good catch, thanks!  Re-reading this, I'm also rewording as:

```
Return a tuple of ``int`` values representing the size of each dimension, which seems a bit more precise.
```

**Comment by bradcray** [comment]
Thanks for the review Lydia!  As an FYI, in re-reviewing, I realized I could/should add return types to these procedures now that they're simpler (not dependent on a config param), so did so.

**Comment by lydia-duncan** [comment]
Sweet, thanks!

**Pull request merged**